src/preact/base-element.js

```diff
@@ -726,6 +726,19 @@ function collectProps(Ctor, element, ref, defaultProps) {
       devAssert(def.attrs);
       value = def.parseAttrs(element);
     }
+    if (def.attrPrefix) {
+      const currObj = {};
+      const attrs = element.attributes;
+      for (let i = 0; i < attrs.length; i++) {
+        const attrib = attrs[i];
+        if (attrib.specified && attrib.name.startsWith(def.attrPrefix)) {
+          currObj[
+            attrib.name.substring(def.attrPrefix.length, attrib.name.length)
```

Comment:
One thing I'm curious about is whether we need to follow `Element.dataset` approach and convert the dash-case to camelcase. Maybe it should be optional. But, e.g. `data-param-special-value` might need to become `specialValue` when passed to the Preact. Is this question already clear from the usecases we have?

Reply:
You're right, the extracted value should be camel cased.